### PR TITLE
Doc/indexing

### PR DESCRIPTION
**Pull request by kavics**
Please review the XML documentation of all files in the Search project and Storage/Search, ContentRepository/Search directories.

### CODE REVIEW

File: src/ContentRepository/Content.cs [hunk 1]
```diff
@@ -749,7 +747,7 @@ private static ContentType ExtendContentType(ISupportsDynamicFields extendedHand
         /// Fully qualified type name is contained by the <see cref="SenseNet.ContentRepository.Schema.ContentType">ContentType</see> named this parameter's value.
         /// </param>
         /// <param name="parent"><see cref="SenseNet.ContentRepository.Storage.Node">ContentHandler</see> instance as a parent in the big tree.</param>
```
Comment (iviczl) [inline src/ContentRepository/Content.cs]: contentTypeName parameter comment is missing

File: src/ContentRepository/Search/TextExtractorsField.cs [hunk 2]
```diff
@@ -5,22 +5,32 @@
...
+    /// Define an inherited <see cref="Field"/> for manafing texte extractor list in the IndexingSettings.
+    /// </summary>
     [ShortName("TextExtractors")]
     [DataSlot(0, RepositoryDataType.NotDefined, typeof(ReadOnlyDictionary<string, ITextExtractor>))]
     [DefaultFieldSetting(typeof(NullFieldSetting))]
     [DefaultFieldControl("SenseNet.Portal.UI.Controls.ShortText")]
     public class TextExtractorsField : Field
     {
+        /// <summary>
+        /// Overridden importer method. Not supported in this field because it has autogeneratd value.
```
Comment (iviczl) [inline src/ContentRepository/Search/TextExtractorsField.cs]: autogeneratEd

File: src/ContentRepository/Search/TextExtractorsField.cs [hunk 3]
```diff
@@ -5,22 +5,32 @@
...
+        /// </summary>
         protected override void ImportData(System.Xml.XmlNode fieldNode, ImportContext context)
         {
             throw new NotSupportedException("The ImportData operation is not supported on TextExtractorsField.");
         }
 
+        /// <summary>
+        /// Overridden exporter method. Does nothing in this field because it has autogeneratd value.
```
Comment (iviczl) [inline src/ContentRepository/Search/TextExtractorsField.cs]: autogeneratEd

File: src/Search/Indexing/IIndexingEngine.cs [hunk 4]
```diff
@@ -4,18 +4,64 @@
...
+        ///   so the order of execution is not important (the inner implementation can be parallelized).
+        /// The method need to be synchronous operation so it returns after full execution.
+        /// Throws an exception if the operation is unsuccessful.
+        /// Note that if the indexing operates uncertainly (means index document existence is not sure),
+        ///   it is strongly recommended to delete the document before addition.
+        ///   In this case the document can be deleted by the VersionId term.
+        /// </summary>
+        /// <param name="deletions">Contains terms that define the documents to delete. Can be null or empty.</param>
+        /// <param name="updates">Contains term-document pairs that define the refreshed items. Can be null or empty.</param>
+        /// <param name="addition">Contains documents to add to index.</param>
```
Comment (iviczl) [inline src/Search/Indexing/IIndexingEngine.cs]: additions would be more appropriate

File: src/ContentRepository/Content.cs (hunk 1 above)
Comment (kavics) [reply src/ContentRepository/Content.cs]: See line 746 (above "parent" param).

File: src/ContentRepository/Search/TextExtractorsField.cs (hunk 2 above)
Comment (kavics) [reply src/ContentRepository/Search/TextExtractorsField.cs]: Done.

File: src/ContentRepository/Search/TextExtractorsField.cs (hunk 3 above)
Comment (kavics) [reply src/ContentRepository/Search/TextExtractorsField.cs]: Done.

File: src/Search/Indexing/IIndexingEngine.cs (hunk 4 above)
Comment (kavics) [reply src/Search/Indexing/IIndexingEngine.cs]: Done.

**Pull request merged**